youtube_dl/extractor/gfycat.py

```diff
@@ -0,0 +1,107 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+
+import datetime
+
+from .common import InfoExtractor
+
+class GfycatIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?gfycat\.com/(?P<id>[^/?#]+)'
+    _TESTS = [{
+        'url': 'http://gfycat.com/DeadlyDecisiveGermanpinscher',
+        'info_dict': {
+            'id':          'DeadlyDecisiveGermanpinscher',
+            'title':       'Ghost in the Shell',
+            'ext':         'mp4',
+            'upload_date': '20140913'
+        }
+    },{
+        'url': 'http://gfycat.com/pleasinghilariouskusimanse',
+        'info_dict': {
+            'id':          'pleasinghilariouskusimanse',
+            'title':       'PleasingHilariousKusimanse',
+            'ext':         'webm',
+            'upload_date': '20150412'
+        }
+    },{
+        'url': 'http://gfycat.com/requiredunkemptbuzzard',
+        'info_dict': {
+            'id':          'requiredunkemptbuzzard',
+            'title':       'Headshot!',
+            'ext':         'gif',
+            'upload_date': '20150130'
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        json     = self._download_json("http://gfycat.com/cajax/get/" + video_id, video_id, 'Downloading video info')['gfyItem']
+        
+        # Title
+        # Use user title first, else fallback to url formated name
+        if json['title']:
+            video_title = json['title']
+        else:
+            video_title = json['gfyName']
+        
+        # Formats
+        # Pref: mp4, webm, gif
+        formats = [{
+            'format_id':  'mp4',
+            'ext':        'mp4',
+            'url':        json['mp4Url'],
+            'width':      json['width'],
+            'height':     json['height'],
+            'fps':        json['frameRate'],
+            'filesize':   json['mp4Size'],
+            'preference': '-1'
```

Comment:
I was not aware of that, I switched `-1` to an integer however rerunning the test I get
```
$ python test/test_download.py TestDownload.test_Gfycat
[Gfycat] DeadlyDecisiveGermanpinscher: Downloading video info
[info] Writing video description metadata as JSON to: DeadlyDecisiveGermanpinscher.info.json
[debug] Invoking downloader on u'http://giant.gfycat.com/DeadlyDecisiveGermanpinscher.gif'
[download] DeadlyDecisiveGermanpinscher.gif has already been downloaded
[download] 100% of 10.00KiB
F
======================================================================
FAIL: test_Gfycat (__main__.TestDownload)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_download.py", line 185, in test_template
    self.assertTrue(os.path.exists(tc_filename), msg='Missing file ' + tc_filename)
AssertionError: Missing file DeadlyDecisiveGermanpinscher.mp4

----------------------------------------------------------------------
Ran 1 test in 0.415s

FAILED (failures=1)
```

It says it's downloading a gif but then says it can't find the mp4. What did I do incorrectly?